### PR TITLE
147394 -  BESS configuration service code upliftment

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
This PBI consists changes of [PBI : 147394](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/147394)

- Upliftment of Configuration Service

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/MacroTransformer.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ string now_Year(Match match)
 
             string nowAddDays_Year(Match match)
             {
-                var dayOffset = offsetCapture(match);
+                int dayOffset = offsetCapture(match);
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/MacroTransformer.cs:30]: Can we declare it at class level e.g. private int dayOffset  = 0; and use it ?
```suggestion
                dayOffset = offsetCapture(match);
```

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs [hunk 2]
```diff
@@ -550,110 +550,105 @@ public enum EventIds
...
-        /// <summary>
-        /// 809109 - Bess config duplicate Name attribute
-        /// </summary>
-        BessConfigDuplicateFileCount = 809109,
-
         /// <summary>
         /// 809110 - Bess config validation summary
         /// </summary>
-        BessConfigValidationSummary = 809110,
+        BessConfigValidationSummary = 809109,
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs:556]: EventId need to be corrected in summary as well for EventIds **809109** to **809128**

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs [hunk 3]
```diff
@@ -123,20 +135,23 @@ public async Task ProcessConfigsAsync()
...
                 {
-                    await CheckConfigFrequencyAndSaveQueueDetails(bessConfigs, salesCatalogueDataResponse.ResponseBody);
+                    var salesCatalogueDataResponse = await salesCatalogueService.GetSalesCatalogueData();
+
+                    await CheckConfigFrequencyAndSaveQueueDetailsAsync(bessConfigs, salesCatalogueDataResponse.ResponseBody);
                 }
 
-                logger.LogInformation(EventIds.BessConfigsProcessingCompleted.ToEventId(), "Bess configs processing completed | _X-Correlation-ID : {CorrelationId}", CommonHelper.CorrelationID);
+                logger.LogInformation(EventIds.BessConfigsProcessingCompleted.ToEventId(), "BESS configs processing completed | _X-Correlation-ID : {CorrelationId}", CommonHelper.CorrelationID);
+                return "BESS configs processing completed";
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs:154]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/MacroTransformer.cs (hunk 1 above)
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/MacroTransformer.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs (hunk 2 above)
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs]: Resolved

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]
minor changes

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/BessBuilderServiceJob.cs [hunk 4]
```diff
@@ -29,21 +29,17 @@ public async Task ProcessQueueMessage([QueueTrigger("%BessStorageConfiguration:Q
...
                     configQueueMessage.Name, CommonHelper.CorrelationID);
-
-                logger.LogInformation(EventIds.BessBuilderServiceCompleted.ToEventId(),
-                    "Bess Builder Service Completed | _X-Correlation-ID : {CorrelationId}", CommonHelper.CorrelationID);
             }
             catch (Exception ex)
             {
-                logger.LogError(EventIds.UnhandledException.ToEventId(), "Exception occurred while processing Bess Builder Service webjob with Exception Message : {Message} | StackTrace : {StackTrace} | _X-Correlation-ID : {CorrelationId}", ex.Message, ex.StackTrace, CommonHelper.CorrelationID);               
+                logger.LogError(EventIds.UnhandledException.ToEventId(), "Exception occurred while processing Bess Builder Service webjob with Exception Message : {Message} | StackTrace : {StackTrace} | _X-Correlation-ID : {CorrelationId}", ex.Message, ex.StackTrace, CommonHelper.CorrelationID);
+                throw;
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/BessBuilderServiceJob.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Extensions/LoggerExtensions.cs [hunk 5]
```diff
@@ -33,14 +33,37 @@ public static class LoggerExtensions
...
-            string messageFormat,
-            Func<Task<TResult>> func,
-            params object[] messageArguments)
+    public static async Task<TResult> LogStartEndAndElapsedTimeAsync<T, TResult>(this ILogger<T> logger,
+    EventIds startEventId,
+    EventIds completedEventId,
+    string startMessageFormat,
+    string completedMessageFormat,
+    Func<Task<TResult>> func,
+    params object[] messageArguments)
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Extensions/LoggerExtensions.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Extensions/LoggerExtensions.cs [hunk 6]
```diff
@@ -33,14 +33,37 @@ public static class LoggerExtensions
...
+                completedMessageFormat + " Elapsed {Elapsed}",
+                messageArguments.Concat(new object[] { stopwatch.Elapsed }).ToArray());
+
+            return result;
+        }
+
+    public static async Task<TResult> LogStartEndAndElapsedTimeAsync<T, TResult>(this ILogger<T> logger,
+    EventIds startEventId,
+    EventIds completedEventId,
+    string startMessageFormat,
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Extensions/LoggerExtensions.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Extensions/LoggerExtensions.cs [hunk 7]
```diff
@@ -49,7 +72,7 @@ public static class LoggerExtensions
 
             stopwatch.Stop();
             logger.LogInformation(completedEventId.ToEventId(),
-                messageFormat + " Elapsed {Elapsed}",
+                startMessageFormat + " Elapsed {Elapsed}",
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Extensions/LoggerExtensions.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/IAzureMessageQueueHelper.cs [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ﻿namespace UKHO.PeriodicOutputService.Common.Helpers;
 public interface IAzureMessageQueueHelper
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/IAzureMessageQueueHelper.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs [hunk 9]
```diff
@@ -551,229 +551,224 @@ public enum EventIds
...
         /// </summary>
-        BessProductTxtAndInfoFolderDeleted = 809143,
+        BESSCleanUpJobRequestStarted = 809143,
 
         /// <summary>
-        /// 809144 - SERIAL.ENC file update operation failed
+        /// 809144 - Bespoke exchange set cleanup web job is completed.
         /// </summary>
-        BessSerialEncUpdateFailed = 809144,
+        BESSCleanUpJobRequestCompleted = 809144,
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs]: Resolved

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

**Review by DanOwenUKHO — APPROVED** [review]

**Pull request merged**